src/ui/app/jsx/repair-schedule-form.jsx

```diff
@@ -0,0 +1,261 @@
+    return {
+      id: this.props.repair ? this.props.repair.id : '',
+      cluster: this.props.repair ? this.props.repair.cluster_name : '',
+      keyspace: this.props.repair ? this.props.repair.keyspace_name : '',
+      owner: this.props.repair ? this.props.repair.owner : '',
+      startTime: this.props.repair ? this.props.repair.next_activation : (this.props.formType === 'schedule' ? moment().toDate() : null),
+      intervalDays: this.props.repair ? this.props.repair.scheduled_days_between : '',
+      tables: this.props.repair ? this.props.repair.column_families : [],
+      tablesBlacklisted: this.props.repair ? this.props.repair.blacklisted_tables : [],
+      nodes: this.props.repair ? this.props.repair.nodes : [],
+      datacenters: this.props.repair ? this.props.repair.datacenters : [],
+      segments: this.props.repair ? this.props.repair.segment_count_per_node : '',
+      parallelism: this.props.repair ? this.props.repair.repair_parallelism : 'PARALLEL',
+      intensity: this.props.repair ? this.props.repair.intensity : '',
+      incrementalRepair: this.props.repair ? this.props.repair.incremental_repair : 'false',
+      repairThreadCount: this.props.repair ? this.props.repair.repair_thread_count : 1,
+      formType: this.props.formType,
+      advancedSettingsOpen: false,
+      valid: undefined  // Indicates if the current state of the form is valid or not
+    }
+  },
+
+  _formSelectChangeHandler: function(valueContext, actionContext) {
+    const field = actionContext.name.split("_")[1];
+    const value = valueContext.value;
+    this._setStateFormChange(field, value);
+  },
+  
+  _formInputChangeHandler: function(e) {
+    const field = e.target.id.split("_")[1];
+    const value = e.target.value;
+    this._setStateFormChange(field, value);
+  },
+
+  _setStateFormChange: function(field, value) {
+    if (!field || !field.length) {
+      return;
+    }
+
+    const changeEvent = {
+      field: field,
+      value: value ? value.toString() : undefined,
+      origValue: this.state && field && this.state[field] ? this.state[field].toString() : undefined
+    }
+
+    const state = this.state;
+    if (field) {
+      state[field] = value;
+    }
+    state.valid = this._validate(state);
+    this.setState(state);
+
+    changeEvent.state = state;
+    if (this.props.onChange) {
+      this.props.onChange(changeEvent);
+    }
+  },
+
+  _validate: function(state) {
+    if (!state) {
+      return false;
+    }
+
+    const clusterInvalid = !state.cluster || !state.cluster.length;
+    const keyspaceInvalid = !state.keyspace || !state.keyspace.length;
+    const ownerInvalid = !state.owner || !state.owner.length;
+    const scheduleTimeInvalid = (state.formType === "schedule" && (!state.startTime || !state.intervalDays)) || (state.formType === "repair" && (state.startTime || state.intervalDays));
+    // TODO What should the rules here be for the the combination of Datacenters and Nodes
```

Comment:
I applied this logic from similar that I found in the add form, but wasn't sure how to treat datacenters and nodes.  The logic elsewhere looked like there were scenarios where you could have one but not the other?  I'll need some background on how to apply the validation here -- I could also simply exclude this logic for now since at the moment the form doesn't allow the user to edit these fields.

Reply:
Since those fields aren't editable (and won't be), my opinion is that you shouldn't bother with those checks.

Reply:
👍 I went ahead and pulled this out.